simpleflow/process/supervisor.py

```diff
@@ -162,15 +162,18 @@ def _handle_sigchld(signum, frame):
             # reliably and determining their state ; we could list children via self._processes
             # but identifying if they're alive is more tricky (???????)
             for child in psutil.Process().children():
-                logger.debug("  child: name=%s pid=%d status=%s" % (child.name(), child.pid, child.status()))
-                if child.status() == psutil.STATUS_ZOMBIE:
-                    logger.debug("  process {} is zombie, will cleanup".format(child.pid))
-                    to_clean = [p for p in self._processes if p.pid == child.pid]
-                    for process in to_clean:
-                        # join process to clean it up
-                        process.join()
-                        # remove the process from self._processes so it will be replaced later
-                        self._processes.remove(process)
+                try:
+                    logger.debug("  child: name=%s pid=%d status=%s" % (child.name(), child.pid, child.status()))
+                    if child.status() == psutil.STATUS_ZOMBIE:
+                        logger.debug("  process {} is zombie, will cleanup".format(child.pid))
+                        to_clean = [p for p in self._processes if p.pid == child.pid]
+                        for process in to_clean:
+                            # join process to clean it up
+                            process.join()
+                            # remove the process from self._processes so it will be replaced later
+                            self._processes.remove(process)
+                except psutil.NoSuchProcess:  # May be untimely deceased
```

Comment:
This change may lead to an inaccurate state in `self._processes`, and ultimately less processes in reality that what we wanted (we compensate lost children in `_start_worker_processes()`, where we compare `len(self._processes)` with the desired `self.nb_children`). I think the disappeared process inside `self._processes` will never be cleaned up.

After a quick test of `psutil.Process` in a console, it seems that:
- `.pid` never raises
- `.name()` is memoized: may raise NoSuchProcess on first invocation
- `.status()` raises NoSuchProcess as soon as the process is not here anymore

So maybe we could instead:
```python
for child in psutil.Process().children():
    try:
        name, status = child.name(), child.status()
    except psutil.NoSuchProcess:
        pass
    ...
```
and adapt the rest so `self._processes` stay the closest possible to reality?

Or we can change how we do this completely. What do you think?